ui/app/templates/components/mount-backend-form.hbs

```diff
@@ -2,100 +2,66 @@
-          {{else}}
+          {{#if (eq @mountType "secret")}}
             {{concat "Enable " typeInfo.displayName " Secrets Engine"}}
+          {{else}}
+            {{concat "Enable " typeInfo.displayName " Authentication Method"}}
           {{/if}}
         {{/let}}
-      {{else if (eq this.mountType "auth")}}
-        Enable an Authentication Method
-      {{else}}
+      {{else if (eq @mountType "secret")}}
         Enable a Secrets Engine
+      {{else}}
+        Enable an Authentication Method
       {{/if}}
     </h1>
   </p.levelLeft>
 </PageHeader>
-<form {{on "submit" (perform this.mountBackend)}}>
-  <div class="box is-sideless is-fullwidth is-marginless">
-    <NamespaceReminder @mode="enable" @noun={{if (eq this.mountType "auth") "Auth Method" "Secret Engine"}} />
-    <MessageError @model={{this.model}} @errorMessage={{this.errorMessage}} @errors={{this.errors}} />
-    {{#if this.showEnable}}
+
+<div class="box is-sideless is-fullwidth is-marginless">
+  <NamespaceReminder @mode="enable" @noun={{if (eq @mountType "secret") "Secret Engine" "Auth Method"}} />
+  <MessageError @errorMessage={{this.errorMessage}} />
+  {{#if @mountModel.type}}
+    <form {{on "submit" (perform this.mountBackend)}}>
       <FormFieldGroups
-        @model={{this.mountModel}}
-        @onChange={{this.onTypeChange}}
+        @model={{@mountModel}}
         @renderGroup="default"
+        @onChange={{this.onTypeChange}}
         @modelValidations={{this.modelValidations}}
         @onKeyUp={{this.onKeyUp}}
       />
-      <FormFieldGroups @model={{this.mountModel}} @onChange={{this.onTypeChange}} @renderGroup="Method Options" />
-    {{else}}
-      {{#each (array "generic" "cloud" "infra") as |category|}}
-        <h3 class="title box-radio-header">
-          {{capitalize category}}
-        </h3>
-        <div class="box-radio-container">
-          {{#each (filter-by "category" category this.mountTypes) as |type|}}
-            <BoxRadio
-              @displayName={{type.displayName}}
-              @type={{type.type}}
-              @glyph={{or type.glyph type.type}}
-              @groupValue={{this.mountModel.type}}
-              @groupName="mount-type"
-              @onRadioChange={{queue (fn (mut this.mountModel.type)) (fn this.onTypeChange "type")}}
-              @disabled={{if type.requiredFeature (not (has-feature type.requiredFeature)) false}}
-              @tooltipMessage={{if
-                (or (eq type.type "transform") (eq type.type "kmip") (eq type.type "keymgmt"))
-                (concat
-                  type.displayName
-                  " is part of the Advanced Data Protection module, which is not included in your enterprise license."
-                )
-                "This secret engine is not included in your license."
-              }}
-            />
-          {{/each}}
+      <FormFieldGroups @model={{@mountModel}} @renderGroup="Method Options" />
```

Comment:
Does `FormFieldGroups` handle the updating the model? So we no longer need to pass `@onChange={{this.onTypeChange}}` ?

Reply:
yes, `onChange` is triggered after the model is set within `FormField` as a callback for any side effects (like validation, for example)